lib/splitclient-rb/engine/matchers/semver.rb

```diff
@@ -0,0 +1,188 @@
+# frozen_string_literal: true
+
+module SplitIoClient
+  class Semver
+    METADATA_DELIMITER = '+'
+    PRE_RELEASE_DELIMITER = '-'
+    VALUE_DELIMITER = '.'
+
+    attr_reader :major, :minor, :patch, :pre_release, :is_stable, :version
+
+    def initialize(version)
+      @major = 0
+      @minor = 0
+      @patch = 0
+      @pre_release = []
+      @is_stable = false
+      @version = ''
+      @metadata = ''
+      parse(version)
+    end
+
+    #
+    # Class builder
+    #
+    # @param version [String] raw version as read from splitChanges response.
+    #
+    # @return [type] Semver instance
+    def self.build(version, logger)
+      new(version)
+    rescue NoMethodError => e
+      logger.error("Failed to parse Semver data, incorrect data type:  #{e}")
+      nil
+    rescue StandardError => e
+      logger.error("Failed to parse Semver data:  #{e}")
+      nil
+    end
+
+    #
+    # Check if there is any metadata characters in version.
+    #
+    # @return [type] String semver without the metadata
+    #
+    def remove_metadata_if_exists(old_version)
+      index = old_version.index(METADATA_DELIMITER)
+      return old_version if index.nil?
+
+      @metadata = old_version[index + 1, old_version.length]
+      old_version[0, index]
+    end
+
+    # Compare the current Semver object to a given Semver object, return:
+    #      0: if self == passed
+    #      1: if self > passed
+    #      -1: if self < passed
+    #
+    # @param to_compare [trype] splitio.models.grammar.matchers.semver.Semver object
+    #
+    # @returns [Integer] based on comparison
+    def compare(to_compare)
+      return 0 if @version == to_compare.version
+
+      # Compare major, minor, and patch versions numerically
+      return compare_attributes(to_compare) if compare_attributes(to_compare) != 0
```

Comment:
you are executing twice `compare_attributes`, eg:

```
# Compare major, minor, and patch versions numerically
result = compare_attributes(to_compare)
return result if result != 0
```

Reply:
will update